controllers/tasks.js

```diff
@@ -142,23 +143,59 @@ const updateTaskStatus = async (req, res, next) => {
   try {
     const taskId = req.params.id;
     const { dev } = req.query;
-    const { id: userId } = req.userData;
+    const { id: userId, username } = req.userData;
     const task = await tasks.fetchSelfTask(taskId, userId);
 
     if (task.taskNotFound) return res.boom.notFound("Task doesn't exist");
     if (task.notAssignedToYou) return res.boom.forbidden("This task is not assigned to you");
-    if (task.taskData.status === "VERIFIED")
+    if (task.taskData.status === "VERIFIED" || req.body.status === "MERGED")
       return res.boom.forbidden("Status cannot be updated. Please contact admin.");
 
-    await tasks.updateTask(req.body, taskId);
+    if (task.taskData.status === "COMPLETED" && req.body.percentCompleted < 100) {
+      if (req.body.status === "COMPLETED" || !req.body.status) {
+        return res.boom.forbidden("Task percentCompleted can't updated as status is COMPLETED");
+      }
+    }
```

Comment:
if the task status is `COMPLETED` and the user is trying to change the `percentCompleted` with a value of less than 100, if so then to check wether the user is also trying to change the status of the task to something else then `COMPLETED` otherwise will give an error.
### Basically the status `COMPLETED` should be there only with percentCompleted as `100`